optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/drools/uni/DroolsGroupingUniConstraintStream.java

```diff
@@ -35,18 +36,29 @@
 import static org.drools.model.PatternDSL.declarationOf;
 import static org.drools.model.PatternDSL.pattern;
 
-public final class DroolsGroupingUniConstraintStream<Solution_, A, GroupKey_>
-        extends DroolsAbstractUniConstraintStream<Solution_, GroupKey_> {
+public final class DroolsGroupingUniConstraintStream<Solution_, A, NewA>
+        extends DroolsAbstractUniConstraintStream<Solution_, NewA> {
```

Comment:
If we start using NewA here instead of GroupKey_, we should do it consistenly in all of ConstraintStreams (including the API and Bavet, not just Drools).
Let's discuss on a bluejeans. (I agree that GroupKey_ isn't perfect, but not sure yet if NewA is the best option). Good to merge this meanwhile - temporarly.